src/hardware/voodoo.cpp

```diff
@@ -929,12 +930,13 @@ struct draw_state
 
 struct triangle_worker
 {
-	bool threads_active, use_threads, disable_bilinear_filter;
+	std::atomic_bool threads_active;
+	bool use_threads, disable_bilinear_filter;
 	UINT16 *drawbuf;
 	poly_vertex v1, v2, v3;
 	INT32 v1y, v3y, totalpix;
 	SDL_sem* sembegin[TRIANGLE_THREADS];
-	volatile bool done[TRIANGLE_THREADS];
+	std::atomic_bool done[TRIANGLE_THREADS];
```

Comment:
An unfortunate side-effect of this change: struct is no longer POD, thus C++ stopped handling copy-assignment for us (*thank you, C++! /s*).

Should be an easy fix though; compiler provides some more info:

> ../src/hardware/voodoo.cpp:7188:43: warning: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘struct triangle_worker’ with no trivial copy-assignment; use value-initialization instead [-Wclass-memaccess]

Reply:
right-o - fixed